components/servo/lib.rs

```diff
@@ -1031,7 +737,7 @@ impl Servo {
     }
 
     fn call_delegate_methods(&self) {
-        let events = self.get_events();
+        let events = std::mem::take(&mut *self.messages_for_embedder.borrow_mut());
```

Comment:
if we update how `receive_messages` and `spin_event_loop`, I think we can also further remove `messages_for_embedder` field.

Reply:
Oh, great point. I've gone ahead an removed `messages_for_embedder` and `receive_messages` entirely and opting for an approach where the Servo instance itself detects when shutdown is complete (rather than relying on the compositor's return value). This is in line with the plans we discussed today about have a more per-WebView compositor architecture.